app/Resources/assets/js/eventshow.js

```diff
@@ -141,3 +142,19 @@ eventmetrics.eventshow.setupWikiInputs = function () {
         });
     });
 };
+
+/**
+ * Model for downloading reports.
+ * The title and links of the model are modified when the user selects a report from the menu.
+ */
+eventmetrics.eventshow.setupReportModal = function () {
+    $('#download-modal').on('shown.bs.modal', function (event) {
+        var modal = $(this),
+            reportLink = $(event.relatedTarget);
+        modal.find('.modal-title').text(reportLink.data('modal-title'));
+        modal.find('.wikitext-link').attr('href', reportLink.data('wikitext-link'));
+        modal.find('.csv-link').attr('href', reportLink.data('csv-link'));
+        // Also append the event status, duplicated from the underlying page.
+        modal.find('.event-status').html($('.event-stats-status').html());
+    })
```

Comment:
The one problem I see here is this happens after the modal is loaded, so the user sees the changes to the content as it happens. Just a little bit weird. Could we hook into `show.bs.modal` instead of `shown`?

Reply:
Oops, that's what I meant to do! Looks much better. Thanks.